### PR TITLE
Exploratory testing fixes

### PR DESCRIPTION
**Pull request by palaciosjeremias**
|Related issue|
|---|
|#1641|

## Description
This PR includes the fixes for the minor problems found during the exploratory testing:
 - Fix for uninitialized variable in **is_valid_file** method.
 - Move helper modules to a **lib** folder.
 - Add **component** to the Mandatory fields.
 - Group name is now based on the README file header.

### CODE REVIEW

File: docs/DocGenerator/DocGenerator.py [hunk 1]
```diff
@@ -12,24 +12,25 @@
 import json
 import yaml
 import ast
-from Config import Config
```
Comment (MiguelazoDS) [inline docs/DocGenerator/DocGenerator.py:15]: We should remove the empty __init__.py file.

File: docs/DocGenerator/DocGenerator.py [hunk 2]
```diff
@@ -50,24 +51,20 @@ def is_valid_folder(self, path):
...
                 return False
         for regex in self.include_regex:
-            match = False
-            if regex.match(path):
-                match = True
-                break
-        if not match:
-            return False
-        return True
+            if regex.match(file):
```
Comment (MiguelazoDS) [inline docs/DocGenerator/DocGenerator.py:65]: Looks good!

File: docs/DocGenerator/lib/CodeParser.py [hunk 3]
```diff
@@ -28,8 +25,8 @@ class CodeParser:
     """
     brief: Class that parses the content of the test files.
     """
-    def __init__(self):
-        self.conf = Config()
+    def __init__(self, config):
```
Comment (MiguelazoDS) [inline docs/DocGenerator/lib/CodeParser.py:28]: I like this!

File: docs/DocGenerator/DocGenerator.py (hunk 1 above)
Comment (palaciosjeremias) [reply docs/DocGenerator/DocGenerator.py:15]: The init.py file is required to convert the lib folder into a python module capable to be imported by the DocGenerator.py module.

Comment (MiguelazoDS) [reply docs/DocGenerator/DocGenerator.py:15]: Good to know that, I thought it was only needed a lib folder.

**Review by MiguelazoDS — APPROVED** [review]

**Pull request merged**